README.md

```diff
@@ -1,2 +1,126 @@
+
+## Using dedicated mkdocs settings
+
+You can override any of the settings from your repo, by placing configurations in the `./github/mkdocs` folder in your repo and documentation in your `docs` folder. For example, if your repository has a `./github/mkocs/mkdocs.yml` this will take precedence over [./.github/actions/mkdocs-pages/mkdocs.yml](./.github/actions/mkdocs-pages/mkdocs.yml). 
+
+Please note that adding new files in `docs` requires changing the `nav` in `./github/mkocs/mkdocs.yml`. When you add to `docs` you will require `./github/mkocs/mkdocs.yml` to add these pages to the navigation see [here](#use-a-docs-directory).
+
+## Merging defaults and dedicated settings
+
+To ease the use of defautls and override the required portions only, the local `./github/mkocs/mkdocs.yml` can use the INHERIT mechanism defined by mkdocs and inherit defaults from the [./.github/actions/mkdocs-pages/mkdocs-default.yml](./.github/actions/mkdocs-pages/mkdocs-default.yml), like this one [./.github/actions/mkdocs-pages/mkdocs.yml](./.github/actions/mkdocs-pages/mkdocs.yml).
+
+For example, to change the site name, the repo using the action should have a file `./.github/mkdocs/mkdocs.yml` with content:
+
+```yaml
+INHERIT: mkdocs-default.yml
+
+site_name: my-site-name
+```
+
+If your pages setup requires additional Python packages, you can specify them in [.github/mkdocs/requirements.txt](.github/mkdocs/requirements.txt) The action will install these dependencies automatically.
+
+# Local Testing
+
+To use this repository locally, just clone the repo and run
+
+```bash
+docker compose up -d
+```
+Then navigate to `localhost:8000` in your web browser to see the docs.
+
+All following changes in the `docs` repo and/or in the `.github/mkdocs` config files (e.g. mkdocs.yaml), will be mirrored on localhost:8000.
+
+
+# Details on the Action
+
+It streamlines the workflow for generating documentation from Markdown files and publishing it to GitHub Pages. It uses the package `mike` for versioning.
+
+This action maintains the project documentation in Markdown files within the repository and automates the documentation generation and deployment process.
+
+
+# Overview
+
+The action is a composite GitHub Action that:
+
+* Checks out the repository code.
+* Copies default documentation templates and configurations to the specified `docs_folder`.
+* Sets up Python and installs necessary dependencies for mkdocs with the SciCat styling.
+* Deploys the generated documentation using `mike`, optionally pushing it to a GitHub Pages branch (gh-pages).
+
+
+# Inputs
+
+The following inputs can be provided to customize the behavior of this action:
+| Input Name | Description | Required | Type | Default |
+| ---------- |----------| -------| ------| ------|
+| GITHUB_TOKEN | GitHub token used for authentication when pushing changes. | true | string | N/A |
+| TAG | Commit, tag, or branch to use for generating GitHub links. | false | string | ${{ github.ref_name }} |
```

Comment:
I think the TAG has the effect also to generate versions dropdown in the doc page

see here:

<img width="733" height="933" alt="Image" src="https://github.com/user-attachments/assets/641ce03e-49d6-4710-93aa-e12ac6e1b985" />